### PR TITLE
Fix: handle empty string in connector comment padding

### PR DESCRIPTION
**Pull request by uncledata**
Problem:
if comment is an empty string pad_comment fails on IndexError on comment[0]. Adding a simple check if comment is not empty

### CODE REVIEW

**Comment by tobymao** [comment]
please add a test case

File: tests/test_generator.py [hunk 1]
```diff
@@ -43,3 +44,12 @@ def test_identify(self):
...
         assert parse_one("X as 1").sql(identify="safe") == 'X AS "1"'
+
+    def test_pad_comment(self):
+        self.assertEqual(Generator().pad_comment(""), "")
+        self.assertEqual(Generator().pad_comment(" leading"), " leading ")
+        self.assertEqual(Generator().pad_comment("trailing "), " trailing ")
+        self.assertEqual(
+            Generator().pad_comment(" leading and trailing "), " leading and trailing "
+        )
+        self.assertEqual(Generator().pad_comment("no padding"), " no padding ")
```
Comment (georgesittas) [inline tests/test_generator.py:55]: Can we instead test this end-to-end by simply adding an example that fails to roundtrip in pretty.sql?

File: tests/test_generator.py [hunk 2]
```diff
@@ -43,3 +44,12 @@ def test_identify(self):
         assert parse_one("X").sql(identify="safe") == "X"
         assert parse_one("x as 1").sql(identify="safe") == '"x" AS "1"'
         assert parse_one("X as 1").sql(identify="safe") == 'X AS "1"'
+
+    def test_pad_comment(self):
+        self.assertEqual(Generator().pad_comment(""), "")
```
Comment (georgesittas) [inline tests/test_generator.py:49]: You don't have to instantiate `Generator` multiple times, just do it once

File: sqlglot/generator.py [hunk 3]
```diff
@@ -647,9 +647,11 @@ def sep(self, sep: str = " ") -> str:
     def seg(self, sql: str, sep: str = " ") -> str:
         return f"{self.sep(sep)}{sql}"
 
-    def pad_comment(self, comment: str) -> str:
-        comment = " " + comment if comment[0].strip() else comment
-        comment = comment + " " if comment[-1].strip() else comment
+    @staticmethod
+    def pad_comment(comment: str) -> str:
```
Comment (georgesittas) [inline sqlglot/generator.py:651]: Nit: I understand this was changed because `self` is not referenced in the body of the method, but note that we don't use `staticmethod`s anywhere else in the code base at the moment, so this creates a small inconsistency.

@tobymao wdyt? Should we refactor other similar methods into `@staticmethod`s?

File: sqlglot/generator.py [hunk 4]
```diff
@@ -648,8 +648,9 @@ def seg(self, sql: str, sep: str = " ") -> str:
         return f"{self.sep(sep)}{sql}"
 
     def pad_comment(self, comment: str) -> str:
-        comment = " " + comment if comment[0].strip() else comment
-        comment = comment + " " if comment[-1].strip() else comment
+        if comment:
```
Comment (georgesittas) [inline sqlglot/generator.py:651]: This makes the `if comment` check in L673 redundant now, let's get rid of it

**Comment by uncledata** [comment]
hm it seems that if moving the check if comment is empty str/None to the method will still cause some issues for some tests:
1. returning an empty string will cause /**/ in situations like:
```sql
select x, --
from foo""",
```
   2. returning None will make the comment become /*None*/
 

 Maybe it's better to add a check to connector_sql and in maybe_comment? I see it's validated like this everywhere else (
```sql
if comment:
```

**Comment by uncledata** [comment]
TL;DR:

Revert my changes in the first place and add if comment check in connector_sql:

```python
                        if comment:
                            op += f" /*{self.pad_comment(comment)}*/"
```

**Comment by georgesittas** [comment]
Sure let's do that

**Review by georgesittas — APPROVED** [review]
LGTM, let's add a test in `test_transpile.py` and I'll get this in
